### PR TITLE
add internal network config to docker-compose file as well

### PR DESCRIPTION
**Pull request by szaimen**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -47,3 +47,16 @@ services:
   #     - ./data:/data
   #     - ./sites:/srv
   #   network_mode: "host"
+
+# # Optional: If you need ipv6, follow step 1 and 2 of https://github.com/nextcloud/all-in-one/blob/main/docker-ipv6-support.md first and then uncomment the below config in order to activate ipv6 for the internal nextcloud-aio network
+# networks:
```
Comment (szaimen) [inline docker-compose.yml:52]: do we need to add network:external somewhere or so?

File: docker-compose.yml [hunk 2]
```diff
@@ -47,3 +47,17 @@ services:
   #     - ./data:/data
   #     - ./sites:/srv
   #   network_mode: "host"
+
+# # Optional: If you need ipv6, follow step 1 and 2 of https://github.com/nextcloud/all-in-one/blob/main/docker-ipv6-support.md first and then uncomment the below config in order to activate ipv6 for the internal nextcloud-aio network
+# # Inspired by https://github.com/mailcow/mailcow-dockerized/blob/master/docker-compose.yml
+# networks:
+#   nextcloud-aio:
+#     driver: bridge
```
Comment (Zoey2936) [inline docker-compose.yml:55]:
```suggestion
#     name: nextcloud-aio
#     driver: bridge
```

File: docker-compose.yml [hunk 3]
```diff
@@ -47,3 +47,17 @@ services:
...
+#     driver_opts:
+#       com.docker.network.bridge.name: nextcloud-aio
+#     enable_ipv6: true
+#     ipam:
+#       driver: default
+#       config:
+#         - subnet: ${IPV4_NETWORK:-172.22.1}.0/24
```
Comment (Zoey2936) [inline docker-compose.yml:62]:
```suggestion
```

Comment (Zoey2936) [reply docker-compose.yml:62]: can be also removed in manual install

File: docker-compose.yml (hunk 2 above)
Comment (Zoey2936) [reply docker-compose.yml:55]: see: https://docs.docker.com/compose/compose-file/compose-file-v3/#name-1

Comment (szaimen) [reply docker-compose.yml:55]: okay! do we want to add external: true to the config or rather not?

File: docker-compose.yml (hunk 3 above)
Comment (szaimen) [reply docker-compose.yml:62]: okay nice!

File: docker-compose.yml (hunk 2 above)
Comment (Zoey2936) [reply docker-compose.yml:55]: I think `external: true` is not needed

Comment (szaimen) [reply docker-compose.yml:55]: apparently rather not

File: docker-compose.yml [hunk 4]
```diff
@@ -47,3 +47,17 @@ services:
...
   #   network_mode: "host"
+
+# # Optional: If you need ipv6, follow step 1 and 2 of https://github.com/nextcloud/all-in-one/blob/main/docker-ipv6-support.md first and then uncomment the below config in order to activate ipv6 for the internal nextcloud-aio network
+# # Inspired by https://github.com/mailcow/mailcow-dockerized/blob/master/docker-compose.yml
+# networks:
+#   nextcloud-aio:
+#     name: nextcloud-aio
+#     driver: bridge
+#     driver_opts:
+#       com.docker.network.bridge.name: nextcloud-aio
```
Comment (Zoey2936) [inline docker-compose.yml:58]: is this needed?

Comment (szaimen) [reply docker-compose.yml:58]: probably not

Comment (Zoey2936) [reply docker-compose.yml:58]:
```suggestion
```

**Comment by Zoey2936** [comment]
can you maybe also add these suggestions to the manual install network config?

**Review by Zoey2936 — APPROVED** [review]

**Comment by szaimen** [comment]
> can you maybe also add these suggestions to the manual install network config?

done with https://github.com/nextcloud/all-in-one/commit/beffb879dc45a74e5f9ad3e8ce8745bd8cfb57d4

**Comment by szaimen** [comment]
Thanks for the review! :)

**Comment by Zoey2936** [comment]
one problem:
- docker-compose only creates the network if at least one service in the container uses the network

**Comment by Zoey2936** [comment]
and:
- maybe the default recommended network should not be the same as mailcows

**Comment by Zoey2936** [comment]
so the mastercontainer would maybe need:
```
    network_mode: bridge
    networks:
      - nextcloud-aio
```
and all compose files (normal & manual install) should use the same ipv6 subnet as the ipv6 docs

**Comment by szaimen** [comment]
> * docker-compose only creates the network if at least one service in the container uses the network

I see. So we would need to add it possinly commented to the mastercontainer. Do you mind creating a PR with that?

> maybe the default recommended network should not be the same as mailcows

What do you mean?

**Comment by szaimen** [comment]
Ah yes, probably we should change that...

**Comment by Zoey2936** [comment]
and something other, does the ipv6 docs use public ipv6 adresses?

**Comment by szaimen** [comment]
Apparently to be private it needs to start with fd:
https://en.m.wikipedia.org/wiki/Private_network#IPv6
Screenshot_20230503_192303_Brave

**Comment by szaimen** [comment]
Lets use fd12:3456:789a:1::/64 and 	fd12:3456:789a:2::/64 then

**Pull request merged**